compiler/testData/diagnostics/tests/smartCasts/varnotnull/leakingLambdaInCalledInPlace.kt

```diff
@@ -0,0 +1,11 @@
+// FIR_IDENTICAL
+fun main() {
+    var p: String?
+    var block: () -> Int = { 1 }
+    p = "2"
+    run {
+        block = { <!SMARTCAST_IMPOSSIBLE!>p<!>.length }
```

Comment:
I don't really like this error here BTW because it says "'p' is a local variable that is captured by a changing closure" but that's wrong.